src/OpenSpaceToolkit/Astrodynamics/Trajectory/CoordinatesBroker.cpp

```diff
@@ -0,0 +1,157 @@
+/// Apache License 2.0
+
+#include <OpenSpaceToolkit/Core/Error.hpp>
+
+#include <OpenSpaceToolkit/Astrodynamics/Trajectory/CoordinatesBroker.hpp>
+
+namespace ostk
+{
+namespace astro
+{
+namespace trajectory
+{
+
+CoordinatesBroker::CoordinatesBroker()
+    : nextCoordinatesSubsetIndex_(0),
+      coordinatesSubsets_({}),
+      coordinatesSubsetsIndexMap_({})
+{
+}
+
+CoordinatesBroker::CoordinatesBroker(const Array<Shared<const CoordinatesSubset>>& aCoordinatesSubsetsArray)
+    : CoordinatesBroker()
+{
+    for (Shared<const CoordinatesSubset> subset : aCoordinatesSubsetsArray)
+    {
+        this->addSubset(subset);
+    }
+}
+
+bool CoordinatesBroker::operator==(const CoordinatesBroker& aCoordinatesBroker) const
+{
+    if (this->getNumberOfCoordinates() != aCoordinatesBroker.getNumberOfCoordinates())
+    {
+        return false;
+    }
+
+    if (this->getNumberOfSubsets() != aCoordinatesBroker.getNumberOfSubsets())
+    {
+        return false;
+    }
+
+    for (const auto& element : this->coordinatesSubsetsIndexMap_)
+    {
+        const String id = element.first;
+        if (!aCoordinatesBroker.hasSubset(id))
+        {
+            return false;
+        }
+
+        if (element.second != aCoordinatesBroker.getSubsetIndex(id))
+        {
+            return false;
+        }
+    }
+
+    return true;
+}
+
+bool CoordinatesBroker::operator!=(const CoordinatesBroker& aCoordinatesBroker) const
+{
+    return !((*this) == aCoordinatesBroker);
+}
+
+Size CoordinatesBroker::getNumberOfCoordinates() const
+{
+    return this->nextCoordinatesSubsetIndex_;
+}
+
+Size CoordinatesBroker::getNumberOfSubsets() const
+{
+    return (size_t)this->coordinatesSubsetsIndexMap_.size();
```

Comment:
```suggestion
    return (Size)this->coordinatesSubsetsIndexMap_.size();
```

It's the same, simply aligning